### PR TITLE
Feat/AN-4120 playwright e2e tests

### PR DESCRIPTION
**Pull request by MGJamJam**
Issue: https://bitmovin.atlassian.net/browse/AN-4120

Work: 
- Adds Playwright e2e tests for plugin.
- Adds default value for percentile value.
- Increases the MinuteThreshold for the AUTO interval calculation to 3h1m

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -60,34 +60,6 @@ jobs:
           version: latest
           args: buildAll
 
-      - name: Check for E2E
```
Comment (Svarozic) [inline .github/workflows/ci.yml:63]: why changes in github workflow files ? Have they not been generated by grafana plugin generator tool ?

File: .gitignore [hunk 2]
```diff
@@ -24,11 +24,15 @@ coverage
 artifacts/
 work/
 ci/
-e2e-results/
-**/cypress/videos
-**/cypress/report.json
```
Comment (Svarozic) [inline .gitignore:29]: Why is "cypress" mentioned in project files ? I thought new plugin has only playwright ? is this just left over from angular grafana plugin ?

File: cypress/integration/01-smoke.spec.ts [hunk 3]
```diff
@@ -1,10 +0,0 @@
-import { e2e } from '@grafana/e2e';
-
-e2e.scenario({
-  describeName: 'Smoke test',
-  itName: 'Smoke test',
-  scenario: () => {
-    e2e.pages.Home.visit();
-    e2e().contains('Welcome to Grafana').should('be.visible');
-  },
-});
```
Comment (Svarozic) [inline cypress/integration/01-smoke.spec.ts:10]: what created this files ?

File: e2e/configurationEditor.spec.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { test, expect } from '@grafana/plugin-e2e';
+import { BitmovinDataSourceOptions } from '../src/types/grafanaTypes';
+
+test('"Save & test" should be successful when configuration is valid', async ({
```
Comment (Svarozic) [inline e2e/configurationEditor.spec.ts:4]: can we split it and create 
- top level `describe('configurationEdtor')`
  - under to have `it('should save and test valid configuration')` (your scenario)
  - and another unsuccessfull scenario `it('should not save invalid configuration')`, where we use use `test-api-key` string and try to save it (do not mock http responses, let backend tell us licenses cant be fetched)

File: package.json [hunk 5]
```diff
@@ -62,11 +61,16 @@
...
+    "@grafana/ui": "10.1.0",
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "tslib": "2.5.3"
   },
+  "resolutions": {
+    "jest/**/string-width": "4.2.3",
+    "isaacs/**/wrap-ansi": "8.1.0",
+    "jest/**/wrap-ansi": "7.0.0"
+  },
```
Comment (Svarozic) [inline package.json:74]: why we need this hardcoded resolutions ? This should have comment at least in git commit message, so in future we can remove it, once we understand why it is there and we can evaluate if it is still needed

File: src/components/ConfigEditor.tsx [hunk 6]
```diff
@@ -59,6 +59,7 @@ export function ConfigEditor(props: Props) {
             value={jsonData.apiKey || ''}
             placeholder="Analytics API Key"
             width={40}
+            id={`config-editor-${props.options.name}_api-key-input`}
```
Comment (Svarozic) [inline src/components/ConfigEditor.tsx:62]: I would still give it another 30mins to investigate, if grafana UI components like here `<Input>` do not support some other attribute for test selectors, because `id` is special and has to be unique, so it would be better to have some `data-testid` / `data-playwrightid` or something else 
I know that `data-testid` is not valid prop and is not present in DOM , but maybe upgrade of grafana UI library, or some other "test attribute" could work ? I would still give it another try

File: src/components/GroupByRow.tsx [hunk 7]
```diff
@@ -41,10 +41,11 @@ type Props = {
   readonly isAdAnalytics: boolean;
   readonly onChange: (newGroupBys: Array<QueryAttribute | QueryAdAttribute>) => void;
   readonly groupBys: Array<QueryAttribute | QueryAdAttribute>;
+  readonly queryEditorId: string;
 };
 
 export function GroupByRow(props: Props) {
-  const paddingTop= props.groupBys.length === 0 ? 4 : 0
+  const paddingTop = props.groupBys.length === 0 ? 4 : 0;
```
Comment (Svarozic) [inline src/components/GroupByRow.tsx:48]: if formatting is wrong, I would create new PR based on `main` and solve it there:
- format project code
- implement "format check" in CI
- merge that PR to `main` and then merge latest `main` here, to reduce code change lines in this PR, because they are not relevant for playwright task

File: src/types/grafanaTypes.ts [hunk 8]
```diff
@@ -52,6 +52,7 @@ export const DEFAULT_QUERY: Partial<BitmovinAnalyticsDataQuery> = {
   filter: [],
   resultFormat: 'time_series',
   interval: 'AUTO',
+  percentileValue: 95,
```
Comment (Svarozic) [inline src/types/grafanaTypes.ts:55]: this looks like fix/feature that has nothing to do with playwright ? could it be done in another PR and explained ?

File: src/utils/intervalUtils.ts [hunk 9]
```diff
@@ -51,7 +51,7 @@ export const calculateQueryInterval = (
   }
 
   const intervalInMilliseconds = endTimestamp - startTimestamp;
-  const minuteIntervalLimitInMilliseconds = 3 * 60 * 60 * 1000; // MINUTE granularity for timeframes below 3h
+  const minuteIntervalLimitInMilliseconds = 3 * 60 * 60 * 1000 + 60000; // MINUTE granularity for timeframes below 3h 1m
```
Comment (Svarozic) [inline src/utils/intervalUtils.ts:54]: This should be done on another PR, I also need context what is this fix for ?

File: .gitignore (hunk 2 above)
Comment (MGJamJam) [reply .gitignore:29]: Cypress is mentioned because grafana just recently deprecated cypress in their plugin development (https://grafana.com/docs/grafana/latest/breaking-changes/breaking-changes-v11-0/#the-grafanae2e-testing-tool-is-deprecated) and back in March when I created the plugin with the generator tool, this generator tool was still using cypress.

File: cypress/integration/01-smoke.spec.ts (hunk 3 above)
Comment (MGJamJam) [reply cypress/integration/01-smoke.spec.ts:10]: The generation tool, see answer above 😊

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MGJamJam) [reply .github/workflows/ci.yml:63]: See answer below. 
But good point, instead of having a seperate e2e workflow I will include this here. I will check how the ci file that is generated today with the tool looks like and copy it over.

File: src/utils/intervalUtils.ts (hunk 9 above)
Comment (MGJamJam) [reply src/utils/intervalUtils.ts:54]: Will create a seperate PR with explaining of context

File: src/types/grafanaTypes.ts (hunk 8 above)
Comment (MGJamJam) [reply src/types/grafanaTypes.ts:55]: True, I just discovered it writing the tests, will create a seperate PR for this change

File: src/components/GroupByRow.tsx (hunk 7 above)
Comment (MGJamJam) [reply src/components/GroupByRow.tsx:48]: Addressed in this PR: https://github.com/bitmovin/analytics-grafana-datasource/pull/94

File: src/types/grafanaTypes.ts (hunk 8 above)
Comment (MGJamJam) [reply src/types/grafanaTypes.ts:55]: Adressed in https://github.com/bitmovin/analytics-grafana-datasource/pull/95

File: src/utils/intervalUtils.ts (hunk 9 above)
Comment (MGJamJam) [reply src/utils/intervalUtils.ts:54]: Here the PR: https://github.com/bitmovin/analytics-grafana-datasource/pull/96

File: e2e/configurationEditor.spec.ts (hunk 4 above)
Comment (MGJamJam) [reply e2e/configurationEditor.spec.ts]: I added the second test but there is no such thing as `describe` in playwright tests.

File: package.json (hunk 5 above)
Comment (MGJamJam) [reply package.json]: After installing playwright I wasn't able to run the jest tests anymore. When doing yarn install I got the warning 
```
warning Pattern ["string-width@^4.1.0"] is trying to unpack in the same destination ".../Library/Caches/Yarn/v6/npm-string-width-cjs-4.2.3-269c7117d27b05ad2e536830a8ec895ef9c6d010-integrity/node_modules/string-width-cjs" as pattern ["string-width-cjs@npm:string-width@^4.2.0"]. This could result in non-deterministic behavior, skipping.
```
and when running the jest tests I got the following error

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /home/runner/work/analytics-grafana-datasource/analytics-grafana-datasource/node_modules/string-width/index.js from /home/runner/work/analytics-grafana-datasource/analytics-grafana-datasource/node_modules/cliui/build/index.cjs not supported.
Instead change the require of index.js in /home/runner/work/analytics-grafana-datasource/analytics-grafana-datasource/node_modules/cliui/build/index.cjs to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/runner/work/analytics-grafana-datasource/analytics-grafana-datasource/node_modules/cliui/build/index.cjs:2[9](https://github.com/bitmovin/analytics-grafana-datasource/actions/runs/10098430816/job/27925459055#step:8:10)1:21)`
```
This is a bug in the cache system of yarn v1. I found two suggested ways to solve this:
1) update yarn to latest version. I tried this, and although the resolution then worked correctly, I didn't manage to get the pregenerated eslint configuration from grafana to work with the updated yarn. OR
2) add those resolutions as suggested [here](https://github.com/cypress-io/cypress/issues/27370) and [here](https://github.com/storybookjs/storybook/issues/22431)

File: src/components/ConfigEditor.tsx (hunk 6 above)
Comment (MGJamJam) [reply src/components/ConfigEditor.tsx]: I investigated it again and found out that for all elements beside the `<Select>` and `<RadioButtonGroup>` the `data-testid` is available. I didn't see it before as I was writing the first test for the `<Select>` elements and then just assumed that the other elements also wouldn't work. So with the last commit I changed all `id`s to `data-testid`s for all elements except the `<Select>` and `<RadioButtonGroup>`.
Tried to find a way to also use `data-testid` for the other elements, but nothing I tried worked. Not entirely sure but I think it works for the other elements because under the hood they are simple HTML elements and inherit HTMLElement Attributes, whereas the other two elements (`<Select>` and `<RadioButtonGroup>`) are React JSX elements.

**Review by Svarozic — APPROVED** [review]
